cmsController.php

```diff
@@ -0,0 +1,191 @@
+                               WHERE `portfolioItems`.`id` = :postData;");
+    $query->bindParam(':postData', $postData);
+    $query->execute();
+    return $query->fetchall();
+}
+
+/**
+ * Gets list of articles from DB
+ *
+ * @param $db PDO to select from
+ * @return array of articles
+ */
+function articleList (PDO $db):array {
+    $query=$db->prepare("SELECT `id`,`name` FROM `articles` WHERE `deleted`=0 ");
+    $query->execute();
+    return $query->fetchall();
+}
+
+/**
+ * Takes selected article POSTdata and selects values from DB based on POST value
+ *
+ * @param $db PDO to select from
+ * @param $postData array userresponse for article to select
+ * @return array
+ */
+function selectArticle (PDO $db, $postData):array {
+    $query = $db->prepare("SELECT `id`,`name`,`description`, `URL`
+                               FROM `articles`
+                               WHERE `id`=:postData;");
+    $query->bindParam(':postData', $postData);
+    $query->execute();
+    return $query->fetchall();
+}
+
+/**
+ * Updates About sections on DB
+ *
+ * @param $postData Array used to test if function should run and informs what content should be added to DB
+ * @param $db to add to
+ */
+function updateAbout (array $postData,PDO $db) {
+    foreach($postData as $name=>$content) {
+        $query = $db->prepare("REPLACE INTO `staticContent`(`name`,`content`) VALUES (?,?)");
+        $query->execute([$name, $content]);
+    }
+}
+
+/**
+ * Updates/adds Portfolio item based on Name
+ *
+ * @param $postData array used to test if function should run and informs what content should be added to DB
+ * @param $db to add to
+ */
+function updatePortfolio (array $postData,PDO $db) {
+        $query = $db->prepare("REPLACE INTO `portfolioItems`(`name`,`description`,`imgRef`,`projURL`,`github`)
+                                     VALUES (:name, :descr, :picRef, :url, :github);");
+        if(array_key_exists('pfTitle',$postData)) {
+            $query->bindValue(':name', $postData['pfTitle']);
+            };
+        if(array_key_exists('pfDesc',$postData)) {
+            $query->bindValue(':descr', $postData['pfDesc']);
+            };
+        if(array_key_exists('pfURL',$postData)) {
+            $query->bindValue(':url', $postData['pfURL']);
+        };
+        if(array_key_exists('githubURL',$postData)) {
+            $query->bindValue(':github', $postData['githubURL']);
+        };
+        if(array_key_exists('picSelect',$postData)) {
+            $query->bindValue(':picRef', $postData['picSelect']);
+        };
+        $query->execute();
+}
+
+/**
+ * Takes list of Image names from DB and returns an array
+ *
+ * @param $db PDO to select from
+ * @return array of images, values
+ */
+function getImgDropDown (PDO $db):array {
+    $query=$db->prepare("SELECT `id`,`name` FROM `images` WHERE `deleted` =0;");
+    if ($query->execute()) {
+        $items=$query->fetchall();
+        return $items;
+    } else {
+        echo 'Image list error';
```

Comment:
it would be good practise here to return an empty array so your still meeting the return type check